src/grammar/dml.rs

```diff
@@ -0,0 +1,130 @@
+use super::{parse_ident, parse_where_clause};
+use crate::grammar::parse_expr;
+use crate::parser::Parser;
+use crate::safe_loop;
+use source_gen::lexer::TokenKind;
+use source_gen::syntax::SyntaxKind;
+use source_gen::T;
+
+#[allow(unused)]
+pub(crate) fn parse_dml(p: &mut Parser) {
+    if p.at(T![delete]) {
+        parse_delete(p);
+    } else {
+        parse_update(p);
+    }
+}
+
+pub(crate) fn parse_delete(p: &mut Parser) {
+    p.start(SyntaxKind::DeleteStmt);
+    p.expect(T![delete]);
+    p.expect(T![from]);
+    parse_ident(p, 1..1);
+    parse_where_clause(p);
+    p.eat(T![;]);
+    p.finish();
+}
+
+pub(crate) fn parse_update(p: &mut Parser) {
+    p.start(SyntaxKind::UpdateStmt);
+    p.expect(T![update]);
+    p.expect_one_of(&[T![unquoted_ident], T![quoted_ident]]);
+    parse_set_clause(p);
+    parse_where_clause(p);
+    p.eat(T![;]);
+    p.finish();
+}
+
+pub(crate) fn parse_set_clause(p: &mut Parser) {
+    p.start(SyntaxKind::SetClause);
+    p.expect(T![set]);
+    safe_loop!(p, {
+        parse_expr(p);
```

Comment:
These have to assignments, not expressions, so we do not allow things such as `UPDATE table SET 1;`.
![image](https://github.com/user-attachments/assets/c96a8e08-e3ca-4e75-a224-ce28f34b29d7)